### ISSUE TITLE
non-elegant error when checking FSLeyes

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#3816]
running master, after installation i get this on a station where X11 is not supported, so i suspect that when checking for fsleyes version we get this error:

~~~
Check ANTs compatibility with OS ...................[OK]
[OK]
Check if figure can be opened with matplotlib.......[FAIL] (Using non-GUI backend 'agg')
Check if figure can be opened with PyQt.............[FAIL] ($DISPLAY not set on X11-supporting system)
Check FSLeyes version...............................Exit code 255 occurred. Please report this issue on SCT GitHub: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues
~~~

is this error (last line) expected? it doesn't seem very elegant. 

considering that non-negligible amount of users do not have fsleyes, maybe we should do something about it?

**Comment by joshuacwnewton** [comment]
This is currently working as written in `sct_check_dependencies`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2ea3643585099860751ce2bf5fb76919dd580f86/spinalcordtoolbox/scripts/sct_check_dependencies.py#L374-L379

This was added in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3428. But, we can definitely change this to something more elegant. 

We could also use this opportunity to rewrite the "FSLeyes not found" message:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2ea3643585099860751ce2bf5fb76919dd580f86/spinalcordtoolbox/scripts/sct_check_dependencies.py#L367-L373

Which has also received criticism in the past for being too verbose/confusing.

**Comment by jcohenadad** [comment]
> This was added in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3428. But, we can definitely change this to something more elegant.

yeah... putting myself in the shoes of an 'average SCT user', that message will not be understood as it should be. Maybe just saying something like "FSLeyes does not seem installed or executable (maybe you are connecting to a remote server)."

**Comment by kousu** [comment]
`fsleyes` has a pip package now. It _should_ be safe to just write it into requirements.txt. I hope that even if it was installed with their `fslinstaller.py` that it would show up in `pip`.

**Comment by jcohenadad** [comment]
i'm not sure we want to force the installation of FSLeyes for our SCT users. Let's discuss that in a meeting.

**Comment by mguaypaq** [comment]
Not to pile on, but another point against `pip install fsleyes`, at least by default: I tried it last week, and it caused my laptop to run full-throttle for an hour while it compiled wxPython. Looking at [the installation instructions](https://open.win.ox.ac.uk/pages/fsl/fsleyes/fsleyes/userdoc/install.html#linux-users), this method also relies on some system-level packages not handled by pip, which apparently were already installed on my system, thankfully.

In short: `pip install fsleyes` isn't lightweight!

**Comment by kousu** [comment]
It's okay on linux if you 'apt install python3-wxpython' first.

I think there's a subtle but important line between the declared dependencies (in setup.py) and the install method we recommend (which could change based on using apt/pip/conda/brew/even nixos).

We can and should declare fsleyes as a dependency, even if we also, say, ship a conda environment.yml or a shell script that runs 'apt install python3-wxpython'.
